cli/cliui/agent.go

```diff
@@ -346,3 +349,42 @@ func PeerDiagnostics(w io.Writer, d tailnet.PeerDiagnostics) {
 		_, _ = fmt.Fprint(w, "✘ Wireguard is not connected\n")
 	}
 }
+
+type ConnDiags struct {
+	Info               *workspacesdk.AgentConnectionInfo
+	PingP2P            bool
+	LocalNetInfo       *tailcfg.NetInfo
+	FetchAgentNetcheck func() (healthsdk.AgentNetcheckReport, error)
+	// TODO: More diagnostics
+}
+
+func ConnDiagnostics(w io.Writer, d ConnDiags) {
+	if d.PingP2P {
+		_, _ = fmt.Fprint(w, "✔ You are connected directly, peer-to-peer (p2p).\n")
+		return
+	}
+	_, _ = fmt.Fprint(w, "❗ You are connected via a DERP relay, not directly, peer-to-peer (p2p).\n")
+
+	if d.Info != nil && d.Info.DisableDirectConnections {
+		_, _ = fmt.Fprint(w, "❗ Your Coder administrator has blocked direct connections.\n")
+		return
+	}
+
+	if d.Info != nil && d.Info.DERPMap != nil && !d.Info.DERPMap.HasSTUN() {
+		_, _ = fmt.Fprint(w, "✘ The workspace agent appears to be unable to reach any STUN servers.\nhttps://coder.com/docs/networking/stun")
```

Comment:
This message should say that the DERP map is not configured to use STUN, which will prevent direct connections from starting outside of local networks.